spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -213,9 +216,11 @@ public long executePartitionedDmlStatement(Statement statement) {
 			SpannerPersistentEntity<T> persistentEntity) {
 		//an entity is eligible if all of the following true:
 		//1. entity has eager loaded properties
-		//2. there are no read options, as they can't be applied to a query
-		//3. key set does not have ranges, as they can't be used in a query
+		//2. entity has "Where" annotation
```

Comment:
Could you explain why can't an entity be eagerly loaded if it is annotated with `@Where`?

Reply:
It can. But when the interleaved property with a list of such entries has an own `@Where` - it should have higher priority than an annotation on a class level

Reply:
I think the `isEligibleForEagerFetch` should be renamed

Reply:
`isEligibleForEagerFetch` is intended to determine if we can and should run `executeReadQueryAndResolveChildren` instead of `readContext.read`, hence the name.

If someone want's to use read and pass read options - we need to call `readContext.read`, we can't just replace it with a call to `executeReadQueryAndResolveChildren` and ignore options. The interleaved children will have to be loaded lazily.

Reply:
ok, I gave a wrong explanation, sorry.
It is not related to interleaved fields.
What if the entity itself has a class annotation `@Where`?
In such a case, we can't use the `readContext.read` to fetch it but instead, we should build an SQL query with "SQL where condition"  from the annotation we have.

Reply:
I see your point. 
In that case will need to modify the logic a little bit.

What we need to do is throw an exception when we have `@Interleved` (eager) children or `@Where` present and options are not null.

We can't just silently ignore the options if we can't pass them. Instead, we should let the user know that something is wrong.

Does it make sense?

Reply:
Options? what options?
How they affect each other?
we have a code
```java
	private <T> boolean isEligibleForEagerFetch(KeySet keys, SpannerReadOptions options,
			SpannerPersistentEntity<T> persistentEntity) {
		return persistentEntity.hasEagerlyLoadedProperties() &&
				persistentEntity.findAnnotation(Where.class) == null &&
				options == null && !keys.getRanges().iterator().hasNext();
	}
```
do you propose to throw an exception here like this?
```java
...
		if (options!=null && (persistentEntity.hasEagerlyLoadedProperties() ||  persistentEntity.getWhere() != null )){
			throw new IllegalArgumentException("bla-bla-bla");
		}

		return persistentEntity.hasEagerlyLoadedProperties() &&
				options == null && !keys.getRanges().iterator().hasNext();
...
```

well, can we better convert the SpannerReadOptions to the SpannerQueryOption and throw an exception when it is impossible?

Reply:
Another idea - the method `readContext.read` is very simple, It has completely another design and another idea with "query" methods and related feature-annotations (Interleaved, Where, whatever else). Can we just completely ignore these annotations for `readContext.read` - I mean for `SpannerOperations#read`? We should just stop the usage of this method in repositories

Reply:
or we can design the `@Where` to be easily converted to the `KeySet`.
@Where` - is a new feature and we can do it as we like.

Reply:
There are 3 separate branches
1. no need for eager or `@Where` - we call `readContext.read`
2. we need eager or `@Where` and options are null - in this case we call `executeReadQueryAndResolveChildren`
3. we need eager or `@Where` and options are not null - we throw an exception

What do you think about that approach?

Reply:
deal